readthedocs/core/signals.py

```diff
@@ -49,6 +49,12 @@ def decide_if_cors(sender, request, **kwargs):  # pylint: disable=unused-argumen
     if request.path_info.startswith('/api/v2/sustainability'):
         return True
 
+    # Don't do domain checking for APIv2 when the Domain is known
+    if request.path_info.startswith('/api/v2/'):
+        domain = Domain.objects.filter(domain__icontains=host)
+        if domain.exists():
```

Comment:
I'm -0 on allowing all API calls to all projects from all Domain's. We should still be mapping the domain -> project, so I think we can just add `/api/v2/` to `WHITELIST_URLS`?

Reply:
How do you connect the `Domain -> Project` for this case? When calling `/api/v2` we don't pass the `project=` query string.

Reply:
One thing we can do here is just allow safe methods for `/api/v2` endpoints: GET and HEAD.